### ISSUE TITLE
what MSE actually does?

### ISSUE THREAD

**Issue by iperov** [haofanwang/accurate-head-pose#4]
part of loss is mse ( predicted * idx_tensor, yaw )
```
idx_tensor = [idx for idx in xrange(66)]
idx_tensor = Variable(torch.FloatTensor(idx_tensor)).cuda(gpu)

yaw_predicted = torch.sum(yaw_predicted * idx_tensor, 1)
loss_reg_yaw = reg_criterion(yaw_predicted, label_yaw_cont)
```

but `torch.sum(yaw_predicted * idx_tensor, 1)` has infinite solutions?
for example
for target angle 45
bins 44 = 0.5 , 46 = 0.5 are correct answer.

so what point to use such approach?

**Comment by haofanwang** [comment]
Hi,

Yes, this situation may happen in theory, but not likely to happen in real if you once tried to print softmax result. The reason why we use such an approach is that we wanna the network learn to get more accurate pose, so we implement an expectation operation (backpropagation through MSE loss) rather than maximum after the softmax. More details you can refer to integral regression. Besides, if the network works well, it can learn to avoid such predicament.

But I don't think this matters, even though you use the softmax classification directly, this question still exists.

**Comment by haofanwang** [comment]
It seems that you have figured your problem out. Closed.
